confgenerator/config.go

```diff
@@ -743,30 +723,53 @@ func (l *Logging) Validate(platform string) error {
 	return nil
 }
 
-func (m *Metrics) Validate(platform string) error {
+func (uc *UnifiedConfig) MetricsReceivers() (map[string]MetricsReceiver, error) {
+	validReceivers := map[string]MetricsReceiver{}
+	for k, v := range uc.Metrics.Receivers {
+		validReceivers[k] = v
+	}
+	if uc.Combined != nil {
+		for k, v := range uc.Combined.Receivers {
+			if _, ok := uc.Metrics.Receivers[k]; ok {
+				return nil, fmt.Errorf("metrics receiver %q has the same name as generic receiver %q", k, k)
```

Comment:
@quentinmit: Can we move this to validation? Then we could remove the returned `error` and not have to check the result every time `MetricsReceivers` is called.